submissions/Eugene-CG/OOP-exercise/index.js

```diff
@@ -0,0 +1,144 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+   Code repository: https://github.com/Eugene-CG/a-tiny-JS-world
+   Web app: https://eugene-cg.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+  constructor(name, saying, gender) {
+    this.name = name;
+    this.saying = saying;
+    this.gender = gender;
+  }
+  getName() {
+    return `name:${this.name}; `;
+  }
+  getSaying() {
+    return `saying:${this.saying}; `;
+  }
+  getGender() {
+    return `gender:${this.gender}; `;
+  }
+  getSpecies() {
+    return `species:${this.species}; `;
+  }
+  getLegs() {
+    return `legs:${this.legs}; `;
+  }
+  getHands() {
+    return `hands:${this.hands}; `;
+  }
+  getPaws() {
+    return `paws:${this.paws}; `;
+  }
+}
+class Animal extends Inhabitant {
+  constructor(name, saying, gender) {
+    super(name, saying, gender);
+    // this.paws = 4;
+    // this.species = "animal";
+  }
+  getProperties() {
+    return (
+      super.getName() +
+      super.getSaying() +
+      super.getGender() +
+      super.getSpecies() +
+      super.getPaws()
+    );
+  }
+}
+class Human extends Inhabitant {
+  constructor(name, saying, gender) {
+    super(name, saying, gender);
+    this.legs = 2;
+    this.hands = 2;
+    this.species = "Homo sapiens";
+  }
+  getProperties() {
+    return (
+      super.getName() +
+      super.getSaying() +
+      super.getGender() +
+      super.getSpecies() +
+      super.getLegs() +
+      super.getHands()
+    );
+  }
+}
+class Cat extends Animal {
+  constructor(name, saying = "Meow", gender) {
```

Comment:
When you want to set some default arguments in the function, write them last in the order. Imagine a situation here, when you wold want to use the `Cat` constructor, but would only want to specify `name` and `gender`. It will not work, since if you just use `Cat("Yennifer", "female")`, `female` is received as `saying` and `gender` is undefined.

Reply:
Maybe

Reply:
I hope

Reply:
> 

:3

Reply:
Оціни мемчик

Reply:
![e4de2dbf2130ab60a627e21b120148a6e6f16fb8991c5b387939478abb6490c1_1](https://user-images.githubusercontent.com/70808337/188323630-411dc224-7307-4701-83b8-dba30a95e469.jpg)

Reply:
XD